### PR TITLE
vimPlugins.nvim-spectre: build spectre_oxi

### PR DESCRIPTION
**Pull request by linsui**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Diff LGTM.
It builds fine on linux (x86 and arm) but not on darwin.
It might be related to [this](https://github.com/nvim-pack/nvim-spectre/issues/185).
Maybe try to add the flags mentioned in the issue description.

**Comment by teto** [comment]
Result of `nixpkgs-review pr 277567` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.nvim-spectre</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -908,9 +908,32 @@
     dependencies = with self; [ nvim-lspconfig ];
   };
 
-  nvim-spectre = super.nvim-spectre.overrideAttrs {
-    dependencies = with self; [ plenary-nvim ];
-  };
+  nvim-spectre = super.nvim-spectre.overrideAttrs
+    (old:
+      let
+        spectre_oxi = rustPlatform.buildRustPackage
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/overrides.nix:914]: make it available in passthru in case one want to change it/override it.

**Review by teto — APPROVED** [review]

**Comment by linsui** [comment]
It still failed on darwin. I don't have darwin to debug it. :shrug:

File: pkgs/applications/editors/vim/plugins/overrides.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/overrides.nix:914]: Good idea !

**Comment by linsui** [comment]
I don't know how I can build it on darwin so I only build it on linux.

**Comment by teto** [comment]
> I don't know how I can build it on darwin so I only build it on linux.

That's fine. Just mark the darwin build broken. 

One question I forgot to ask: is spectre_oxi mandatory now or it remains an optional feature you just want to use ? in which case it could be interesting to look at the closure size. (e.g. `nix path-info -Sh .#vimPlugins.nvim-spectre` before and after)

**Comment by linsui** [comment]
> That's fine. Just mark the darwin build broken.

Then I guess the darwin users can't even use the plugin without spectre_oxi?

> One question I forgot to ask: is spectre_oxi mandatory now or it remains an optional feature you just want to use ? in which case it could be interesting to look at the closure size. (e.g. nix path-info -Sh .#vimPlugins.nvim-spectre before and after)

It's optional. The closure increased from 32.8M to 44.4M.

**Comment by GaetanLepage** [comment]
The plugin now builds fine on all platforms !

**Pull request merged**